### ISSUE TITLE
Psalm type IssuerMetadataObject is too strict

### ISSUE THREAD

**Issue by Jean85** [facile-it/php-jose-verifier#4]
I just upgraded to 0.3, and PHPStan is complaining about this:

``` ------ -----------------------------------------------------------------------                                                                                                                            
  46     Parameter #1 $issuerMetadata of method                                                                                                                                                            
         Facile\JoseVerifier\AbstractTokenVerifierBuilder<Facile\JoseVerifier\                                                                                                                             
         AbstractTokenVerifier>::setIssuerMetadata() expects array('issuer' =>                                                                                                                             
         mixed, 'authorization_endpoint' => mixed, ?'token_endpoint' => mixed,                                                                                                                             
         ?'userinfo_endpoint' => mixed, 'jwks_uri' => mixed,                                                                                                                                               
         ?'registration_endpoint' => mixed, ?'scopes_supported' => array<int,                                                                                                                              
         mixed>, 'response_types_supported' => array<int, mixed>&nonEmpty,                                                                                                                                 
         ...), array('issuer' => string, 'jwks_uri' => string) given.                                                                                                                                      
 ------ -----------------------------------------------------------------------                                                                                                                            
```
TL;DR: `setIssuerMetadata()` expects IssuerMetadataObject, which requires (omitting nullable fields):
```
array{
	issuer: non-empty-string,
	authorization_endpoint: non-empty-string,
	jwks_uri: non-empty-string,
	response_types_supported: non-empty-list<OpenIdResponseType>,
	id_token_signing_alg_values_supported: non-empty-list<non-empty-string>,
}
```
This doesn't seem true, since I'm able to call it with just `issuer` and `jwks_uri`. Psalm doesn't complain, but it seems pretty strange to me.

In any case, those types seems too restrictive to me..
